### MGT451 - Assignment #5 ~ Brier Score of Forecaster

**Author**: Sid Bhatia

**Date**: October 19th, 2023

**Pledge**: I pledge my honor that I have abided by the Stevens Honor System.

**Professor**: Dr. Jordan Suchow

In this assignment, you will compute the Brier score of a forecaster.

1. Select a forecasting problem (e.g., the event of whether a friend can correct translate a given English word into French).

The forecasting problem revolves around predicting whether it will rain the following day (binary outcome: 1 for rain, 0 for no rain) based on daily weather forecasts.

2. Find a friend, family member, or classmate to help you out. Either you or your partner can serve as the forecaster.

In this scenario, my friend will serve as the forecaster, providing daily weather forecasts.

3. On each round, the forecaster will provide a forecast (i.e., a probability that the event will occur). Record the forecast and the outcome (1 when it happens, 0 when it does not). Repeat for a reasonable number of events.

In [5]:
import pandas as pd
import numpy as np

# Data Forecasts vs. Actual Outcomes
forecasts = [0.3, 0.6, 0.2, 0.8, 0.4, 0.7, 0.1]
actual_outcomes = [0, 1, 1, 1, 0, 1, 0]  # Actual outcomes (0 for no rain, 1 for rain)

# Create a DataFrame to store the data
data = {'Day': list(range(1, len(forecasts) + 1)),
        'Forecasted Probability': forecasts,
        'Actual Outcome': actual_outcomes}
df = pd.DataFrame(data)

4. Compute the forecaster's Brier score: the mean squared error between the forecast and the outcome.

In [6]:
# Calculate the Brier Score
brier_score = np.mean((df['Forecasted Probability'] - df['Actual Outcome'])**2)

5. Create a 1-page description of what you did, including a description of the forecasting problem (the event), a table of the results, and the computed Brier score. Did they do well?

In [7]:
# Display the table and Brier Score
print("Table of Results:")
print(df)
print(f"\nBrier Score: {brier_score:.2f}")

Table of Results:
   Day  Forecasted Probability  Actual Outcome
0    1                     0.3               0
1    2                     0.6               1
2    3                     0.2               1
3    4                     0.8               1
4    5                     0.4               0
5    6                     0.7               1
6    7                     0.1               0

Brier Score: 0.17


**Assessment**:

The forecaster demonstrated reasonably good performance in predicting rain based on daily weather forecasts. The Brier score of 0.17 indicates that, on average, the forecasts were fairly close to the actual outcomes. While there is always room for improvement, especially on days with unexpected weather patterns, the forecaster's predictions can be considered reliable for practical purposes.

**Conclusion**:

In conclusion, this assignment successfully assessed the forecasting accuracy of a friend or partner using the Brier score. The forecaster's predictions for rain were reasonably accurate, as evidenced by the Brier score of 0.17. This score suggests that the forecasts provided valuable information for decision-making regarding weather-related activities.